lightning/src/events/mod.rs

```diff
@@ -1582,6 +1582,55 @@ pub enum Event {
 		/// onion messages.
 		peer_node_id: PublicKey,
 	},
+	/// Indicates that a funding transaction constructed via interactive transaction construction for a
+	/// channel is ready to be signed by the client. This event will only be triggered
+	/// if at least one input was contributed by the holder and needs to be signed.
+	///
+	/// The transaction contains all inputs provided by both parties  along with the channel's funding
+	/// output and a change output if applicable.
+	///
+	/// No part of the transaction should be changed before signing as the content of the transaction
+	/// has already been negotiated with the counterparty.
+	///
+	/// Each signature MUST use the SIGHASH_ALL flag to avoid invalidation of the initial commitment and
+	/// hence possible loss of funds.
+	///
+	/// After signing, call [`ChannelManager::funding_transaction_signed`] with the (partially) signed
+	/// funding transaction.
+	///
+	/// Generated in [`ChannelManager`] message handling.
```

Comment:
Could you add a "Failure Behavior and Persistence" section as is done for other events?